### PR TITLE
Overhaul organization and add new entries

### PR DESCRIPTION
**Pull request by sjsyrek**
I copy edited the writing and grammar, alphabetized all the entries, added back links to the TOC, added a section with links to FP libraries for JavaScript, rewrote the Applicative Functor entry, and added an entry for Algebraic Data Types. I hope this isn’t too much all at once! Actually, I want to add more, but I figured I should submit less at one time. Please review in the event I got something wrong.

### CODE REVIEW

**Comment by jethrolarson** [comment]
Thanks for the contributions here. Some of these changes would be immediately mergable if it they weren't all in the same pr

**Comment by hemanth** [comment]
Thanks for the wonderful contrib, as @jethrolarson mentioned if you could change the existing PR it would be great!

**Comment by sjsyrek** [comment]
Sure! Yeah, I can see I got a little carried away this time—sorry about that. Any specific suggestions for how to break things up?

File: readme.md [hunk 1]
```diff
@@ -1,54 +1,180 @@
...
 *This is a [WIP](https://github.com/hemanth/functional-programming-jargon/issues/20); please feel free to send a PR ;)*
 
 Where applicable, this document uses terms defined in the [Fantasy Land spec](https://github.com/fantasyland/fantasy-land)
 
+## Functional programming libraries and projects for JavaScript
+
+Practical examples of popular and production-ready FP libraries, FP-inspired projects, and other learning resources:
+
+* [ramda](https://github.com/ramda/ramda)
+* [lodash](https://github.com/lodash/lodash)
```
Comment (hemanth) [inline readme.md]: We can change this to `loadsh-fp`

File: readme.md [hunk 2]
```diff
@@ -1,54 +1,180 @@
...
+* [underscore.js](https://github.com/jashkenas/underscore)
```
Comment (hemanth) [inline readme.md]: underscore and ramda in the same list ;)?

Comment (sjsyrek) [inline readme.md]: We should be agnostic?

File: readme.md (hunk 1 above)
Comment (sjsyrek) [inline readme.md]: Why? I'm not sure I follow. Also, I'm splitting up my pull requests now. Should I wait for each merge to master before making further pull requests? Or make them all at once? I don't want to create merge conflicts on account of my own edits!

File: readme.md (hunk 2 above)
Comment (hemanth) [inline readme.md]: Agree, there is this `lodash-fp` as well.

**Comment by hemanth** [comment]
?

File: readme.md (hunk 1 above)
Comment (hemanth) [inline readme.md]: > I don't want to create merge conflicts on account of my own edits!

Heh heh, that would be interesting, let us see how good we can manage this.

**Pull request closed without merging**